aws_lambda_builders/workflows/nodejs_npm/utils.py

```diff
@@ -48,3 +64,124 @@ def abspath(self, path):
+        """
+
+        if rel_package_path.startswith("file:"):
+            rel_package_path = rel_package_path[5:].strip()
+
+        package_path = osutils.abspath(osutils.joinpath(parent_package_path, rel_package_path))
+
+        if not osutils.dir_exists(scratch_dir):
+            osutils.mkdir(scratch_dir)
+
+        if output_dir is None:
+            # TODO: get a higher level output_dir to keep process locals between jobs
+            output_dir = osutils.joinpath(artifacts_dir, "@aws_lambda_builders_local_dep")
+            if not osutils.dir_exists(output_dir):
+                osutils.mkdir(output_dir)
+            top_level = True
+        else:
+            tarfile_name = subprocess_npm.run(["pack", "-q", package_path], cwd=scratch_dir).splitlines()[-1]
+            tarfile_path = osutils.joinpath(scratch_dir, tarfile_name)
+
+            LOG.debug("NODEJS extracting child dependency for recursive dependency check")
+
+            osutils.extract_tarfile(tarfile_path, artifacts_dir)
+
+            top_level = False
+
+        local_manifest_path = osutils.joinpath(artifacts_dir, "package", "package.json")
+        local_dependencies = DependencyUtils.get_local_dependencies(local_manifest_path, osutils)
+        for (dep_name, dep_path) in local_dependencies.items():
+            dep_scratch_dir = osutils.joinpath(scratch_dir, str(abs(hash(dep_name))))
+
+            # TODO: if dep_scratch_dir exists (anywhere up path), it means we've already processed it this round, skip
+
+            dep_artifacts_dir = osutils.joinpath(dep_scratch_dir, "unpacked")
+
+            LOG.debug("NODEJS packaging dependency, %s, from %s to %s", dep_name, parent_package_path, output_dir)
+
+            dependency_tarfile_path = DependencyUtils.package_local_dependency(
+                package_path, dep_path, dep_artifacts_dir, dep_scratch_dir, output_dir, osutils, subprocess_npm
+            )
+
+            packaged_dependency_tarfile_path = osutils.joinpath(output_dir, osutils.filename(dependency_tarfile_path))
+            osutils.copy_file(dependency_tarfile_path, output_dir)
+
+            LOG.debug("NODEJS packed localized child dependency to %s", packaged_dependency_tarfile_path)
+
+            LOG.debug("NODEJS updating package.json %s", local_manifest_path)
+
+            DependencyUtils.update_manifest(local_manifest_path, dep_name, packaged_dependency_tarfile_path, osutils)
+
+        if not top_level:
+            localized_package_dir = osutils.joinpath(artifacts_dir, "package")
+
+            LOG.debug("NODEJS repackaging child dependency")
+
+            tarfile_name = subprocess_npm.run(
+                ["pack", "-q", localized_package_dir], cwd=localized_package_dir
+            ).splitlines()[-1]
+
+            return osutils.joinpath(localized_package_dir, tarfile_name)
+
+    @staticmethod
+    def update_manifest(manifest_path, dep_name, dependency_tarfile_path, osutils):
```

Comment:
Do you think you could update this method to not use a backup file?
As we are deleting it at the end, it seems unnecessary.
Either opening the file in `r+` mode (but you will have to `truncate()` it) or open in `r` mode, process and reopen in `w` mode to overwrite it. 
It should be as easy as setting your three with/if/with blocks on the same indentation.